apps/studio/src/components/PageEditor/FileAttachment.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState } from "react"
+import { FormControl, Skeleton, Text } from "@chakra-ui/react"
+import { Attachment, FormErrorMessage } from "@opengovsg/design-system-react"
+
+import { MAX_PDF_FILE_SIZE_BYTES } from "~/features/editing-experience/components/form-builder/renderers/controls/constants"
+import { useUploadAssetMutation } from "~/hooks/useUploadAssetMutation"
+import { getPresignedPutUrlSchema } from "~/schemas/asset"
+
+interface FileAttachmentProps {
+  setHref: (href: string) => void
+  siteId: number
+  error?: string
+  setError: (error: string) => void
+  clearError
+}
+
+export const FileAttachment = ({
+  setHref,
+  siteId,
+  error,
+  setError,
+  clearError,
+}: FileAttachmentProps) => {
+  const [file, setFile] = useState<File | undefined>(undefined)
+  // TODO: Add a mutation for deletion next time of s3 resources
+  const { mutate: uploadFile, isLoading } = useUploadAssetMutation({
+    siteId,
+  })
+
+  useEffect(() => {
+    // NOTE: The outer link modal uses this to disable the button
+    if (isLoading) setHref("")
+  }, [isLoading, setHref])
```

Comment:
do we need this to be a `useEffect`?

Reply:
yes sadly. this is because the `useEffect` sync the `isLoading` with the external state. because the `mutate` runs as a side-effect, we need something to sync the `loading` state to whether the button is disabled